docs/fundamentals/networking/networking-telemetry.md

```diff
@@ -0,0 +1,547 @@
+
+This is currently the most accurate way to measure the duration of a given request.
+
+```c#
+public sealed class KestrelTelemetryConsumer : IKestrelTelemetryConsumer
+{
+    private static readonly AsyncLocal<DateTime?> _startTimestamp = new();
+    private readonly ILogger<KestrelTelemetryConsumer> _logger;
+
+    public KestrelTelemetryConsumer(ILogger<KestrelTelemetryConsumer> logger) => _logger = logger;
+
+    public void OnRequestStart(DateTime timestamp, string connectionId, string requestId, string httpVersion, string path, string method)
+    {
+        _startTimestamp.Value = timestamp;
+    }
+
+    public void OnRequestStop(DateTime timestamp, string connectionId, string requestId, string httpVersion, string path, string method)
+    {
+        if (_startTimestamp.Value is { } startTime)
+        {
+            var elapsed = timestamp - startTime;
+            _logger.LogInformation("Request {requestId} to {path} took {elapsedMs} ms", requestId, path, elapsed.TotalMilliseconds);
+        }
+    }
+}
+```
+
+### Measuring the latency of a .NET reverse proxy
+
+Applicable if you have a reverse proxy that receives inbound requests via Kestrel and makes outbound requests via HttpClient (e.g. [YARP]).
+
+Time from receiving request headers until we sent out the headers to the backend server.
+
+```c#
+public sealed class InternalLatencyMonitor : IKestrelTelemetryConsumer, IHttpTelemetryConsumer
+{
+    private record RequestInfo(DateTime StartTimestamp, string RequestId, string Path);
+
+    private static readonly AsyncLocal<RequestInfo> _requestInfo = new();
+    private readonly ILogger<InternalLatencyMonitor> _logger;
+
+    public InternalLatencyMonitor(ILogger<InternalLatencyMonitor> logger) => _logger = logger;
+
+    public void OnRequestStart(DateTime timestamp, string connectionId, string requestId, string httpVersion, string path, string method)
+    {
+        _requestInfo.Value = new RequestInfo(timestamp, requestId, path);
+    }
+
+    public void OnRequestHeadersStop(DateTime timestamp)
+    {
+        if (_requestInfo.Value is { } requestInfo)
+        {
+            var elapsed = (timestamp - requestInfo.StartTimestamp).TotalMilliseconds;
+            _logger.LogInformation("Internal latency for {requestId} to {path} was {duration} ms", requestInfo.RequestId, requestInfo.Path, elapsed);
+        }
+    }
+}
+```
+
+[distributed tracing documentation]: ../../core/diagnostics/distributed-tracing.md
+[EventCounters]: ../../core/diagnostics/event-counters.md
+[well-known counters]: ../../core/diagnostics/available-counters.md
+[dotnet-counter docs]: ../../core/diagnostics/dotnet-counters.md
+[dotnet-trace docs]: ../../core/diagnostics/dotnet-trace.md
+[dotnet-trace analysis docs]: ../../core/diagnostics/dotnet-trace.md#view-the-trace-captured-from-dotnet-trace
+[PerfView]: https://github.com/microsoft/perfview
+[AppInsights EventCounters docs]: /azure/azure-monitor/app/eventcounters
```

Comment:
What should the relative link look like for this one?
I want to link to https://learn.microsoft.com/azure/azure-monitor/app/eventcounters

Reply:
You have it correctly.